include/render.h

```diff
@@ -2,7 +2,8 @@
 #define MAKO_RENDER_H
 
 struct mako_state;
+struct mako_surface;
 
-int render(struct mako_state *state, struct pool_buffer *buffer, int scale);
+int render(struct mako_state *state, struct mako_surface *surface, struct pool_buffer *buffer, int scale);
```

Comment:
Since the state is already stored in the `mako_surface` struct, no need to have it as a separate argument anymore